sdk/src/Core/Amazon.Runtime/Credentials/AWSCredentials.cs

```diff
@@ -35,11 +35,9 @@ public abstract class AWSCredentials : BaseIdentity
         /// </summary>
         protected virtual void Validate() { }
 
-#if AWS_ASYNC_API
```

Comment:
Why are we removing `AWS_ASYNC_API`? (By the way I think this constant is redundant since every target supports async but it's in a lot of other places too)

Reply:
I removed it based on discussion with @normj since it is now redundant. And based on discussion, the current scope of removal was limited to credential providers.